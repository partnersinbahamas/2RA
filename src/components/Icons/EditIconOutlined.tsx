import React from 'react';
import PropTypes from 'prop-types';
import { BLACK } from '../Colors/colors';
import Svg from './Svg';
import { TIcon } from '.';

/**
 * @deprecated - Use EditIcon size=LARGE instead
 */
const EditIconOutlined: React.FC<TIcon> = ({
  className,
  color = BLACK,
  size = 'LARGE',
  style,
}) => {
  const sizeData: any = {
    LARGE: {
      path: 'M21.8438 0C22.5625 0 23.2812 0.28126 23.9375 0.9375L27.0938 4.09376C27.6562 4.6875 28 5.4375 28 6.25C28 6.96876 27.75 7.6875 27.0938 8.34376L10.0625 25.3125C9.6875 25.6875 9.3125 25.9688 8.625 26.125L1.25 27.9375C1.15625 27.9688 1.03125 27.9688 0.9375 27.9688C0.40625 27.9688 0 27.5313 0 27.0313C0 26.9375 0 26.8438 0.03125 26.75L1.875 19.3125C2 18.7188 2.25 18.3438 2.6875 17.9063L19.6875 0.9375C20.25 0.375 21.0312 0 21.8438 0ZM3.5 20.9688L2.34375 25.625L6.96875 24.4688L3.5 20.9688ZM18.4688 4.96876L4.40625 19L8.9375 23.5938L23 9.53126L18.4688 4.96876ZM21.8125 2.03126C21.5625 2.03126 21.3125 2.125 21.0938 2.34376L19.875 3.5625L24.4375 8.15626L25.6875 6.90626C25.8438 6.75 26 6.5 26 6.1875C26 5.96876 25.9062 5.75 25.6875 5.5L22.5312 2.34376C22.3438 2.15626 22.0938 2.03126 21.8125 2.03126Z',
      viewBox: '-2 -2 32 32',
    },
    MEDIUM: {
      path: 'M15.47 0c.56 0 1.1.21 1.5.63l2.3 2.28c.45.47.73 1.06.73 1.7 0 .56-.21 1.14-.73 1.66L7.53 18.03c-.28.29-.63.54-.96.6l-5.3 1.33c-.08.02-.15.02-.24.02a.94.94 0 01-.7-.3c-.22-.21-.38-.52-.31-.8l.02-.12 1.27-5.12c.1-.4.28-.75.54-1.03L13.73.73C14.2.26 14.83 0 15.47 0zM3.4 13.83l-.07.07c-.03.03-.05.12-.07.21l-.87 3.48 3.43-.82a.5.5 0 00.23-.1l.1-.1-2.75-2.74zm9.02-8.95l-7.57 7.57 2.7 2.72 7.59-7.59-2.72-2.7zm2.98-2.84c-.1 0-.19.03-.26.12l-1.32 1.32 2.75 2.74 1.32-1.34a.3.3 0 00.09-.23c0-.1-.05-.21-.14-.3l-2.18-2.19c-.07-.1-.17-.12-.26-.12z',
      viewBox: '-2 -2.02 24 24',
    },
    SMALL: {
      path: 'M12.27 0c.49 0 .97.16 1.33.49l1.82 1.82c.36.35.55.94.55 1.49 0 .52-.2 1.07-.55 1.46l-9.19 9.12c-.32.3-.58.4-1.1.52l-3.93 1c-.1.04-.2.04-.3.04a.78.78 0 01-.6-.26c-.17-.16-.3-.42-.3-.68v-.16l.97-3.93c.2-.72.3-.85.62-1.17L10.74.62c.43-.43.98-.62 1.53-.62zM9.41 4.8l-5.84 5.75 1.79 1.82 5.84-5.78L9.4 4.8zm2.83-2.85L10.8 3.38l1.82 1.81 1.42-1.42-1.81-1.82z',
      viewBox: '-2.015 -2.03 20 20',
    },
  };
  return (
    <Svg
      xmlns="http://www.w3.org/2000/svg"
      x="0px"
      y="0px"
      viewBox={sizeData[size].viewBox}
      style={style}
      className={className}
    >
      <path fill={color} d={sizeData[size].path} />
    </Svg>
  );
};

EditIconOutlined.propTypes = {
  className: PropTypes.string,
  color: PropTypes.string,
  size: PropTypes.oneOf(['LARGE', 'MEDIUM', 'SMALL']),
  style: PropTypes.any,
};

export default EditIconOutlined;
