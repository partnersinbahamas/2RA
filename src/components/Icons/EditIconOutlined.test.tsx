import React from 'react';
import { render } from '@testing-library/react';
import EditIconOutlined from './EditIconOutlined';

test('EditIconOutlined', () => {
  const { container } = render(<EditIconOutlined />);
  expect(container).toMatchSnapshot();
});
