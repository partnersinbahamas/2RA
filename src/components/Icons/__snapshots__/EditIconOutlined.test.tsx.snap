// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditIconOutlined 1`] = `
<div>
  <svg
    class="sc-beySPh jroDkm"
    viewBox="-2 -2 32 32"
    x="0px"
    xmlns="http://www.w3.org/2000/svg"
    y="0px"
  >
    <path
      d="M21.8438 0C22.5625 0 23.2812 0.28126 23.9375 0.9375L27.0938 4.09376C27.6562 4.6875 28 5.4375 28 6.25C28 6.96876 27.75 7.6875 27.0938 8.34376L10.0625 25.3125C9.6875 25.6875 9.3125 25.9688 8.625 26.125L1.25 27.9375C1.15625 27.9688 1.03125 27.9688 0.9375 27.9688C0.40625 27.9688 0 27.5313 0 27.0313C0 26.9375 0 26.8438 0.03125 26.75L1.875 19.3125C2 18.7188 2.25 18.3438 2.6875 17.9063L19.6875 0.9375C20.25 0.375 21.0312 0 21.8438 0ZM3.5 20.9688L2.34375 25.625L6.96875 24.4688L3.5 20.9688ZM18.4688 4.96876L4.40625 19L8.9375 23.5938L23 9.53126L18.4688 4.96876ZM21.8125 2.03126C21.5625 2.03126 21.3125 2.125 21.0938 2.34376L19.875 3.5625L24.4375 8.15626L25.6875 6.90626C25.8438 6.75 26 6.5 26 6.1875C26 5.96876 25.9062 5.75 25.6875 5.5L22.5312 2.34376C22.3438 2.15626 22.0938 2.03126 21.8125 2.03126Z"
      fill="#000000"
    />
  </svg>
</div>
`;
